### PR TITLE
Better error messages from Final and NoPublicConstructor

### PR DESCRIPTION
**Pull request by njsmith**
This is a pretty trivial change, but I noticed it while working on
gh-1496 so I figured I'd just fix it.

### CODE REVIEW

File: trio/_util.py [hunk 1]
```diff
@@ -216,7 +216,9 @@ def __new__(cls, name, bases, cls_namespace):
         for base in bases:
             if isinstance(base, Final):
                 raise TypeError(
-                    "`%s` does not support subclassing" % base.__name__
+                    "the {!r} class does not support subclassing".format(
+                        base.__name__
```
Comment (oremanj) [inline trio/_util.py:220]: Do we want to do `{__module__}.{__qualname__}` for maximum comprehensibility?

**Comment by oremanj** [comment]
LGTM, one optional suggestion

File: trio/_util.py (hunk 1 above)
Comment (njsmith) [reply trio/_util.py:220]: Good idea! Done.

**Comment by njsmith** [comment]
(Also, whoa, I just realized we can use f-strings now!)

**Pull request merged**